src/components/classrooms/StudentsTable.js

```diff
@@ -0,0 +1,352 @@
+  static propTypes = {
+    classroom: PropTypes.instanceOf(Map).isRequired,
+    classes: PropTypes.shape({
+      root: PropTypes.string.isRequired,
+      paper: PropTypes.string.isRequired,
+      table: PropTypes.string.isRequired,
+    }).isRequired,
+    t: PropTypes.func.isRequired,
+    dispatchDeleteUserInClassroom: PropTypes.func.isRequired,
+    userId: PropTypes.string.isRequired,
+  };
+
+  state = (() => {
+    const { classroom } = this.props;
+    const rows = classroom.get('users').map(createStudentRow);
+
+    return {
+      order: TABLE_ORDER.ASC,
+      orderBy: USERNAME_HEAD_CELL_ID,
+      selected: [],
+      page: 0,
+      dense: false,
+      rowsPerPage: 5,
+      rows,
+    };
+  })();
+
+  buildHeadCells = () => {
+    const { t } = this.props;
+    return [
+      createHeadCell(USERNAME_HEAD_CELL_ID, t('Username')),
+      createHeadCell(OPERATIONS_HEAD_CELL_ID, t('Operations')),
+      // todo : add header cell per space
+    ];
+  };
+
+  handleRequestSort = (event, property) => {
+    const { orderBy, order } = this.state;
+    const isAsc = orderBy === property && order === TABLE_ORDER.ASC;
+    this.setState({
+      order: isAsc ? TABLE_ORDER.DESC : TABLE_ORDER.ASC,
+      orderBy: property,
+    });
+  };
+
+  handleSelectAllClick = event => {
+    if (event.target.checked) {
+      const { rows } = this.state;
+      const newSelecteds = rows.map(n => n[USERNAME_HEAD_CELL_ID]);
+      this.setState({ selected: newSelecteds });
+      return;
+    }
+    this.setState({ selected: [] });
+  };
+
+  handleClick = (event, name) => {
+    const { selected } = this.state;
+    const selectedIndex = selected.indexOf(name);
+    let newSelected = [];
+
+    if (selectedIndex === -1) {
+      newSelected = newSelected.concat(selected, name);
+    } else if (selectedIndex === 0) {
+      newSelected = newSelected.concat(selected.slice(1));
+    } else if (selectedIndex === selected.length - 1) {
+      newSelected = newSelected.concat(selected.slice(0, -1));
+    } else if (selectedIndex > 0) {
+      newSelected = newSelected.concat(
+        selected.slice(0, selectedIndex),
+        selected.slice(selectedIndex + 1)
+      );
+    }
+
+    this.setState({ selected: newSelected });
+  };
+
+  handleChangePage = (event, page) => {
+    this.setState({ page });
+  };
+
+  handleChangeRowsPerPage = event => {
+    const rowsPerPage = parseInt(event.target.value, 10);
+    this.setState({ page: 0, rowsPerPage });
+  };
+
+  handleChangeDense = event => {
+    this.setState({ dense: event.target.checked });
+  };
+
+  handleDeleteUser = row => {
+    const { id: userId } = row;
+    const {
+      dispatchDeleteUserInClassroom,
+      classroom,
+      userId: teacherId,
+    } = this.props;
+    dispatchDeleteUserInClassroom({
+      username: row[USERNAME_HEAD_CELL_ID],
+      userId,
+      teacherId,
+      classroomId: classroom.get('id'),
+    });
+  };
+
+  renderDenseSwitch = () => {
```

Comment:
Do we really need the dense switch? I think it might clutter and b/c it is handled in the state, it is not persistent. Maybe we can kill it?